src/vs/vscode.d.ts

```diff
@@ -2736,14 +2787,8 @@ declare module 'vscode' {
 
 		/**
 		 * Provide completion items for the given position and document.
-		 *
-		 * @param document The document in which the command was invoked.
-		 * @param position The position at which the command was invoked.
-		 * @param token A cancellation token.
-		 * @return An array of completions, a [completion list](#CompletionList), or a thenable that resolves to either.
-		 * The lack of a result can be signaled by returning `undefined`, `null`, or an empty array.
 		 */
-		provideCompletionItems(document: TextDocument, position: Position, token: CancellationToken): ProviderResult<CompletionItem[] | CompletionList>;
+		provideCompletionItems: CompletionItemProvider.ProvideCompletionItems | CompletionItemProvider.ProvideCompletionItemsForContext;
 
```

Comment:
I'll bring it in todays standup but my personal preference is to have the OR-types defined in place and not as separate interface. That gives them somehow to much importance...

Reply:
Ok. We have discussed this in great depth during the standup. We got creative and came up with 5 or 6 ways how to implement this. For the sake of simplicity, less API changes, and for the sake a sane calling side (the function.length trick has a dark side) we decided to ditch the "cancellation token comes last"-rule in *this* case.